packages/deepbook/sources/balance_manager.move

```diff
@@ -48,67 +66,96 @@ module deepbook::balance_manager {
-    public fun authorize_trader(
-        balance_manager: &mut BalanceManager,
-        authorize_address: address,
-        ctx: &mut TxContext,
-    ) {
+    /// Returns the balance of a Coin in an balance_manager.
+    public fun balance<T>(balance_manager: &BalanceManager): u64 {
+        let key = BalanceKey<T> {};
+        if (!balance_manager.balances.contains(key)) {
+            0
+        } else {
+            let acc_balance: &Balance<T> = &balance_manager.balances[key];
+            acc_balance.value()
+        }
+    }
+
+    /// Mint a `TradeCap`, only owner can mint a `TradeCap`.
+    public fun mint_trade_cap(balance_manager: &mut BalanceManager, ctx: &mut TxContext): TradeCap {
+        balance_manager.validate_owner(ctx);
+        assert!(balance_manager.allow_listed.size() < MAX_TRADE_CAPS, EMaxTradeCapsReached);
+
+        let id = object::new(ctx);
+        balance_manager.allow_listed.insert(id.to_inner());
+
+        TradeCap {
+            id,
+            balance_manager_id: object::id(balance_manager),
+        }
+    }
+
+    /// Revoke a `TradeCap`. Only the owner can revoke a `TradeCap`.
+    public fun revoke_trade_cap(balance_manager: &mut BalanceManager, trade_cap_id: &ID, ctx: &TxContext) {
         balance_manager.validate_owner(ctx);
-        assert!(balance_manager.authorized_traders.size() < MAX_TRADERS, EMaxTraderReached);
-        assert!(
-            !balance_manager.authorized_traders.contains(&authorize_address),
-            ETraderAlreadyInList,
-        );
-        balance_manager.authorized_traders.insert(authorize_address);
+
+        assert!(balance_manager.allow_listed.contains(trade_cap_id), ETradeCapNotInList);
+        balance_manager.allow_listed.remove(trade_cap_id);
     }
 
-    /// Remove an authorized_trader. Only the owner can remove.
-    public fun remove_trader(
-        balance_manager: &mut BalanceManager,
-        trader_address: address,
-        ctx: &TxContext,
-    ) {
+    /// Generate a `TradeProof` by the owner. The owner does not require a capability
+    /// and can generate TradeProofs without the risk of equivocation.
+    public fun generate_proof_as_owner(balance_manager: &mut BalanceManager, ctx: &TxContext): TradeProof {
         balance_manager.validate_owner(ctx);
-        assert!(balance_manager.authorized_traders.contains(&trader_address), ETraderNotInList);
-        balance_manager.authorized_traders.remove(&trader_address);
+
+        TradeProof {
+            balance_manager_id: object::id(balance_manager),
+            trader: ctx.sender(),
+        }
     }
 
-    /// Deposit funds to an balance_manager. Only owner can deposit.
-    public fun deposit<T>(self: &mut BalanceManager, coin: Coin<T>, ctx: &mut TxContext) {
-        self.validate_owner(ctx);
-        self.deposit_protected(coin.into_balance(), ctx);
+    /// Generate a `TradeProof` with a `TradeCap`.
+    /// Risk of equivocation since `TradeCap` is an owned object.
```

Comment:
I don't think we should mention these as being a "risk", sounds a bit like a bad thing even though it exists for a good reason!